package/kernel/mwlwifi/patches/001-Fix-compilation-warning-with-64-bit-system.patch

```diff
@@ -119,6 +119,19 @@ Signed-off-by: Christian Marangi <ansuelsmth@gmail.com>
 
 --- a/debugfs.c
 +++ b/debugfs.c
+@@ -364,10 +364,10 @@ static ssize_t mwl_debugfs_info_read(str
+ 			 "-----------------------=>  address| address|qlen|fw_desc_cnt\n");
+ 		spin_lock_irqsave(&pcie_priv->tx_desc_lock, flags);
+ 		len += scnprintf(p + len, size - len,
+-				"wcb_base0   : %x => %8x|%8p|%4d|%d\n", get_hw_spec->wcb_base0, *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base0)),skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
++				"wcb_base0   : %x => %8x|%8p|%4u|%d\n", get_hw_spec->wcb_base0, le32_to_cpu(get_hw_spec->wcb_base0),&get_hw_spec->wcb_base0,skb_queue_len(&pcie_priv->txq[0]),pcie_priv->fw_desc_cnt[0]);
+ 		for(i = 0; i < SYSADPT_TOTAL_TX_QUEUES - 1; i++)
+ 			len += scnprintf(p + len, size - len,
+-				"wcb_base[%2d]: %x => %8x|%8p|%4d|%d\n", i, get_hw_spec->wcb_base[i], *((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),(void *)*((unsigned int *)le32_to_cpu(get_hw_spec->wcb_base[i])),skb_queue_len(&pcie_priv->txq[i + 1]),pcie_priv->fw_desc_cnt[i + 1]);
++				"wcb_base[%2d]: %x => %8x|%8p|%4u|%d\n", i, get_hw_spec->wcb_base[i], le32_to_cpu(get_hw_spec->wcb_base[i]),&get_hw_spec->wcb_base[i],skb_queue_len(&pcie_priv->txq[i + 1]),pcie_priv->fw_desc_cnt[i + 1]);
```

Comment:
The prints are different. The old one is printing the content of `get_hw_spec->wcb_base[i]`, se the `*` which references the address. The new one is only printing the address.

I am not sure if this debug information is needed.

Reply:
I doubt it.

Not sure what to do here. It's throwing a bunch of -Wint-to-pointer-cast

Reply:
Yeah

```
include <stdio.h>
#include <stdint.h>

int main()
{
        unsigned int i = 5;
        printf("MANGIX: %x\n", *(unsigned int *)i);
}
```

segfaults.

Reply:
OK, looks like your program is not allowed to access memory address 5, but what's the point here?
There’s no reason to assume that the content of `get_hw_spec->wcb_base[i]` yields the same error.

Reply:
with godbolt it segfaults. See https://godbolt.org/z/e4qTrqoeK

Tempest in a teapot honestly. This code is in debug output. I don't think much care should be made with it.

Reply:
@Ansuel funny enough I have an mvebu device running Debian. Segfaults there too.